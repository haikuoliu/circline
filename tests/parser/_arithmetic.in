do 1 + 2.1
do 1 - 2.1
do 1 * 2.1
do 1 / 2.1
do 1 % 2.1
do 1 ** 2.1
do -42
do 1 + -43
do 1 * 2 + 3 ** 4
do 1 / 2 % 3 % 4
do 1 + 2 - 3 / 4
do 1 * (2 + 3)
