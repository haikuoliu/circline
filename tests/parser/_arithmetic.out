[
	Do(Binop(Num_lit(1), Add, Num_lit(2.1))) ;
	Do(Binop(Num_lit(1), Sub, Num_lit(2.1))) ;
	Do(Binop(Num_lit(1), Mult, Num_lit(2.1))) ;
	Do(Binop(Num_lit(1), Div, Num_lit(2.1))) ;
	Do(Binop(Num_lit(1), Mod, Num_lit(2.1))) ;
	Do(Binop(Num_lit(1), Pow, Num_lit(2.1))) ;
	Do(Unop(Sub, Num_lit(42))) ;
	Do(Binop(Num_lit(1), Add, Unop(Sub, Num_lit(43)))) ;
	Do(
		Binop(
			Binop(Num_lit(1), Mult, Num_lit(2)),
			Add,
			Binop(Num_lit(3), Pow, Num_lit(4))
		)
	) ;
	Do(
		Binop(
			Binop(
				Binop(Num_lit(1), Div, Num_lit(2)),
				Mod,
				Num_lit(3)
			),
			Mod,
			Num_lit(4)
		)
	) ;
	Do(
		Binop(
			Binop(Num_lit(1), Add, Num_lit(2)),
			Sub,
			Binop(Num_lit(3), Div, Num_lit(4))
		)
	) ;
	Do(
		Binop(
			Num_lit(1),
			Mult,
			Binop(Num_lit(2), Add, Num_lit(3))
		)
	)
]
